rules/solid/src/Rector/Variable/MoveVariableDeclarationNearReferenceRector.php

```diff
@@ -90,27 +98,36 @@ public function refactor(Node $node): ?Node
     /**
      * Find the first node within the same method being a usage of the assigned variable,
      * but not the original assignment itself.
+     *
+    * @param ClassMethod|Function_|Class_|Namespace_|Closure $parentScope
      */
-    private function findFirstVariableUsageInScope(Variable $variable, Assign $assign, Node $parentScope): ?Node
+    private function findFirstVariableUsageInScope(Variable $desiredVariable, Assign $assign, Node $parentScope): ?Variable
     {
-        return $this->betterNodeFinder->findFirst(
+        $desiredVariableName = $this->getName($desiredVariable);
+
+        $firstVariableUsage = $this->betterNodeFinder->findFirst(
             (array) $parentScope->getStmts(),
-            function (Node $node) use ($variable, $assign): bool {
-                return $this->isVariable($node) &&
-                    ! $this->isOriginalAssign($node, $assign) &&
-                    $this->betterStandardPrinter->areNodesEqual($node, $variable);
```

Comment:
I'm not sure where the problem was. Maybe here. This is very hard to read for me, so I used early-returns. It might have fixed the issue.

Reply:
I tested to use `areNodesEqual()` instead of `isName()`, but it had no effect.